### PR TITLE
7 add votes

### PR DESCRIPTION
**Pull request by kerberos16**
Articles can be voted up and down

Once a vote up is given, the button is disabled. If the user has made a mistake, he/she can click on the vote down button. This will remove the vote and reset both buttons.
The votes should be saved and visible if the user adds a vote on the articles page and then when viewing an individual article

-flaws:
- if a user gives a vote on the articles page, then goes to the individual article page, the vote button will clickable and not disabled

### CODE REVIEW

File: src/components/article-card/article-card.jsx [hunk 1]
```diff
@@ -1,32 +1,78 @@
...
 
 
-const ArticleCard = ({ article }) => {
+const ArticleCard = ({ article, setArticlesList }) => {
+
+  const [selectedArticle, setSelectedArticle] = useState(article)
+  const [disableAdd, setDisableAdd] = useState(false)
+  const [disableRemove, setDisableRemove] = useState(false)
+
+  const addVote = () => {
```
Comment (dappernerddesigns) [inline src/components/article-card/article-card.jsx]: I don't understand why this is here when you have an api file with the exact same function, which you have imported.

File: src/components/article-card/article-card.jsx [hunk 2]
```diff
@@ -1,32 +1,78 @@
...
+    .catch((err) => console.log(err, "Something went wrong, cannot add vote!"))
+
+    setSelectedArticle((currentSelectedArticle) => {
+        return {...currentSelectedArticle, votes: currentSelectedArticle.votes + 1}
+      })
+      setDisableAdd(true)
+  }
+
+  const removeVote = () => {
+    axios
```
Comment (dappernerddesigns) [inline src/components/article-card/article-card.jsx]: I also don't understand why this is here. You can use one api call to amend the votes by number sent. Exactly how you have it in the back end. Adding 1 or adding -1 will achieve the upvote and downvote.

**Review by dappernerddesigns — CHANGES_REQUESTED** [review]
I am struggling to understand why a user can upvote or downvote an article before they have read it. Adding upvote and downvote buttons to the article card does not make sense. 

Currently when upvoting the button is correctly disabled. If I then downvote, I can continue to downvote infinitely. This is the same both on article list and on single article.

With no internet connection I can continue to upvote and downvote. Maybe let the user know there is an issue here?

Make sure to remove console logs!!

You have repeated your api calls several times. You have created an api file, which is great and really useful. But then you go on to re write the same functions inside your components which is repeating yourself and not using the imported functions.

I'd like you to tidy up your api calls and address your disabling buttons. I'd like either both to work well, or just let the user upvote/downvote as much as you like. When complete, please resubmit your PR.

**Comment by k-huggs** [comment]
remove your console.logs
also you disable the buttons once you vote, but what if you change your mind and want to switch how you voted?

**Pull request merged**